apps/antalmanac/src/components/RightPane/SectionTable/SectionTableBody.tsx

```diff
@@ -349,97 +362,119 @@ interface SectionTableBodyProps {
-                coursePaneEvent.endTime = translatedTimeString.endTime;
-            }
+    // Because the idiot who set up state management didn't use an actual library solution,
+    // we need to attach ***memoized*** listeners to the store like bufoons.
 
-            for (const calendarEvent of calendarEventTimes) {
-                // Check if there is day overlap
-                if (!coursePaneEvent?.day?.includes(calendarEvent?.day)) {
-                    continue;
-                }
-
-                // Then, IF the course doesn't ( start AND end BEFORE) AND doesn't ( start AND end AFTER), it does conflict!
-                const happensBefore =
-                    coursePaneEvent.startTime <= calendarEvent.startTime &&
-                    coursePaneEvent.endTime <= calendarEvent.startTime;
-                const happensAfter =
-                    coursePaneEvent.startTime >= calendarEvent.endTime &&
-                    coursePaneEvent.endTime >= calendarEvent.endTime;
-
-                // If neither happensBefore or happensAfter is true, set scheduleConflict to true
-                if (!happensBefore && !happensAfter) {
-                    setScheduleConflict(true);
-                    return;
-                }
-            }
-            setScheduleConflict(false);
-            return;
+    useEffect(() => {
+        RightPaneStore.on('columnChange', handleColumnChange);
+        return () => {
+            RightPaneStore.removeListener('columnChange', handleColumnChange);
         };
+    }, [handleColumnChange]);
 
-        const updateCourseState = () => {
-            updateHighlight();
-            checkAndDisplayScheduleConflict();
+    useEffect(() => {
+        AppStore.on('addedCoursesChange', updateHighlight);
+        AppStore.on('currentScheduleIndexChange', updateHighlight);
+
+        return () => {
+            AppStore.removeListener('addedCoursesChange', updateHighlight);
+            AppStore.removeListener('currentScheduleIndexChange', updateHighlight);
         };
+    }, [updateHighlight]);
 
-        updateCourseState();
-        AppStore.on('addedCoursesChange', updateCourseState);
-        AppStore.on('currentScheduleIndexChange', updateCourseState);
+    useEffect(() => {
+        AppStore.on('addedCoursesChange', updateCalendarEvents);
+        AppStore.on('currentScheduleIndexChange', updateCalendarEvents);
 
         return () => {
-            AppStore.removeListener('addedCoursesChange', updateCourseState);
-            AppStore.removeListener('currentScheduleIndexChange', updateCourseState);
+            AppStore.removeListener('addedCoursesChange', updateCalendarEvents);
+            AppStore.removeListener('currentScheduleIndexChange', updateCalendarEvents);
         };
-    }, [section.sectionCode, term]); //should only run once on first render since these shouldn't change.
+    }, [updateCalendarEvents]);
+
+    const scheduleConflict = useMemo(() => {
```

Comment:
This value is derived from other stateful values, so it shouldn't be recorded with `useState` and `useEffect`. If it's declared as an un-memoized value, then it's re-calculated on every single render anyways. It's memoized because that's literally the correct way to write this dumb shit.